src/Symfony/Component/Ldap/Connection/Connection.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Connection;
+
+use Symfony\Component\Ldap\Exception\ConnectionException;
+use Symfony\Component\Ldap\Exception\LdapException;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class Connection implements ConnectionInterface
+{
+    private $bound = false;
+    private $connection;
+    private $config;
+
+    /**
+     * Constructor.
+     *
+     * @param array $options
+     */
+    public function __construct(array $config = array())
+    {
+        if (!extension_loaded('ldap')) {
+            throw new LdapException('The ldap module is needed.');
+        }
+
+        $resolver = new OptionsResolver();
```

Comment:
Do you really need to use the OptionsResolver component ?
It looks like a simple array_replace is enough.

It will reduce the number of dependencies on the LDAP component;
Therefore, the OptionsResolver is very slow